### ISSUE TITLE
Jenkins Gatling plugin build with multiple reports in different stages fails to render second report

### ISSUE THREAD

**Issue by bobonaut** [gatling/gatling#3816]
Running two gatling tests in jenkins and archiving them using the 'gatlingArchive()' step results in two gatling reports which are archived in Jenkins, however only one report is accessible.

- Jenkins Gatling plugin version 1.26.

Explanations:
We currently have a pipeline set up where we need to run a performance test in several stages.
This is a basic gatling performance test which runs perfectly individually without an issue.

The setup of our jenkins pipeline looks something as following:
```groovy
#!groovy

pipeline {

    agent none

    options {
        timestamps()

        stages {
            stage("Running performance test") {
                steps {
                    withMaven(maven: 'maven-3') {
                        sh 'mvn clean install -Pperformance-test'

                        gatlingArchive()
                    }
                }
            }
            
            stage("Running performance test 2") {
                steps {
                    withMaven(maven: 'maven-3') {
                        sh 'mvn clean install -Pperformance-test'

                        gatlingArchive()
                    }
                }
            }
        }
    }
}
```

With this setup we run two stages separately and publish the gatling reports using gatlingArchive() jenkins plugin function.
This works for both reports, however we can only render the initial report, trying to render second report results in an NPE, with the following stacktrace

```
java.lang.NullPointerException
	at io.gatling.jenkins.ReportRenderer.<init>(ReportRenderer.java:51)
	at io.gatling.jenkins.GatlingBuildAction.getReport(GatlingBuildAction.java:82)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:710)
	at org.kohsuke.stapler.Function$MethodFunction.invoke(Function.java:396)
	at org.kohsuke.stapler.Function$InstanceFunction.invoke(Function.java:408)
	at org.kohsuke.stapler.MetaClass$4.doDispatch(MetaClass.java:280)
	at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:58)
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:747)
Caused: javax.servlet.ServletException
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:797)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:878)
	at org.kohsuke.stapler.MetaClass$9.dispatch(MetaClass.java:456)
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:747)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:878)
	at org.kohsuke.stapler.MetaClass$9.dispatch(MetaClass.java:456)
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:747)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:878)
	at org.kohsuke.stapler.MetaClass$4.doDispatch(MetaClass.java:280)
	at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:58)
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:747)
... ommitted the rest
```

This seems to be an issue with how the build simulations are stored in the build.xml of a specific build and the way the report is rendered.

**Comment by bobonaut** [comment]
@slandelle We've done some digging into the issue and the only problem seems to be in the representation of the reports. The actual files and data are being uploaded to jenkins and stored in the build.xml. 
I think the fix should be trivial, I'm currently figuring out how to run everything locally so i can fix it myself.

**Comment by slandelle** [comment]
Great! Feel free to send a PR when you can.

**Comment by bobonaut** [comment]
@slandelle i've submitted a PR: https://github.com/jenkinsci/gatling-plugin/pull/20. 
This fixes the issue of multiple simulations in different steps.

The plugin is designed to use a single "BuildAction" and  using "gatlingArchive" pipeline command in multiple steps adds a new build action every time. Instead it is possible to reuse an existing action, this lets the reports to be added and looked up in a proper manner.

**Comment by ccedric** [comment]
Hi @bobonaut 

Thanks for reporting and contributing, I took a look at your pull request.

I think that our plugin is working correctly in this case, it is just a bad usage of the plugin due to an incomplete documentation.

The Gatling plugin for Jenkins is meant to be a post-build action, run once for the build. It archives any Gatling tests which were not yet archived.

In your case, you should call gatlingArchive() only once at the end of your build. Your two reports will be archived, and available from the UI:

![jenkins-gatling-multiple-reports](https://user-images.githubusercontent.com/7644199/68200026-5fa50f00-ffbf-11e9-8437-1199c14c9a90.png)

Do you confirm that it fixes your issue?

**Comment by bobonaut** [comment]
Hi @ccedric,

Thank you for your analysis, your approach is correct only for a limited number of use cases.
It would work if everything is executed on one agent and the build is only cleaned up at the end.

When the pipeline is designed to be modular and run on different agents in single/parallel stages after which the workspace is cleaned up, then the approach you're proposing doesn't fix the issue.

With the design of jenkins pipelines where a user can run things on different environments which can even be dynamic (destroyed after a stage) it is impossible to collect multiple reports from multiple places afterward.

**Comment by ccedric** [comment]
Thanks for explaining, I understand your use case.
I'll test your PR, thanks for contributing.

**Comment by slandelle** [comment]
Released 1.2.7.
